### PR TITLE
domainforlocalgateway is now dynamic

### LINKED ISSUE #76: Pull domainforlocalgateway from upstream ingress rather than hard coding value
see func domainForLocalGateway in ingress.go.

### PR DESCRIPTION
**Pull request by AngeloDanducci**
# Changes

- domainForLocalGateway in ingress.go based on the ingress

/kind enhancement

Fixes #76

### CODE REVIEW

**Comment by AngeloDanducci** [comment]
One point of discussion is this does not use the `-new` created ingresses as the source of truth (as mentioned in the TODO) but depends on the initial configuration otherwise set to a sensible default.

To my knowledge the ingress LB domain doesn't change when a new ingress is created but is based on which ingress class is used.  Given that other pieces rely on the initial config to be correct this seems an acceptable dynamic approach.

If we close this as is I will open a new issue for ambassador/contour support, discussion around that is in the mentioned issue.

**Comment by AngeloDanducci** [comment]
@maximilien could use a review

File: pkg/reconciler/ingress/ingress.go [hunk 1]
```diff
@@ -192,12 +205,18 @@ func markIngressReady(ingress *v1alpha1.Ingress) {
...
 }
 
-// TODO(bvennam) we need to pull this from the upstream ingress that is create "-new"
 func domainForLocalGateway(ingressName string, isPrivate bool) string {
-	if isPrivate {
-		return privateLBDomain
+	if LBDomain, ok := loadBalancers[strings.Split(ingressName, ".")[0]]; ok {
+		if isPrivate {
+			return LBDomain.Private
+		}
```
Comment (maximilien) [inline pkg/reconciler/ingress/ingress.go:212]: this if statement can be rewritten in a easier fashion.

Comment (maximilien) [reply pkg/reconciler/ingress/ingress.go:212]: It’s confusing right now since checking for isPrivate twice

Comment (maximilien) [reply pkg/reconciler/ingress/ingress.go:212]: Split into helper funcs if needed

**Review by maximilien — CHANGES_REQUESTED** [review]

File: pkg/reconciler/ingress/ingress.go (hunk 1 above)
Comment (AngeloDanducci) [reply pkg/reconciler/ingress/ingress.go]: Split each case (known good from available, or default if none known good are found) into a helper function.

File: pkg/reconciler/ingress/ingress.go [hunk 2]
```diff
@@ -192,14 +205,29 @@ func markIngressReady(ingress *v1alpha1.Ingress) {
 	ingress.Status.MarkNetworkConfigured()
 }
 
-// TODO(bvennam) we need to pull this from the upstream ingress that is create "-new"
 func domainForLocalGateway(ingressName string, isPrivate bool) string {
+	// checks for a valid domain in the list of load balancers
+	if LBDomain, ok := loadBalancers[strings.Split(ingressName, ".")[0]]; ok {
```
Comment (maximilien) [inline pkg/reconciler/ingress/ingress.go:210]: What if not "ok"? Need a check

**Review by maximilien — COMMENTED** [review]
Need check for not OK and also tests

File: pkg/reconciler/ingress/ingress.go (hunk 2 above)
Comment (AngeloDanducci) [reply pkg/reconciler/ingress/ingress.go:210]: If not "ok" this uses the default load balancer domain.

Added testing for malformed entries, this now accepts only istio/kourier as valid ingress and load balancer domains.

File: cmd/consumer/main.go [hunk 3]
```diff
@@ -61,6 +61,7 @@ func consumeEvent(event cloudevents.Event) error {
 	if err != nil {
 		return fmt.Errorf("problem calling url: %w", err)
 	}
+	log.Printf("Received response: %s\n", resp.Status)
```
Comment (maximilien) [inline cmd/consumer/main.go:64]: This looks like debug left in code...

File: pkg/reconciler/ingress/ingress.go [hunk 4]
```diff
@@ -43,33 +44,44 @@ const (
...
+
 // ReconcileKind implements Interface.ReconcileKind.
 func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) reconciler.Event {
 	logger := logging.FromContext(ctx)
 	ingressClass := os.Getenv(ingressClassName)
 
-	if strings.HasSuffix(ingressClass,ingressSuffix) {
-		logger.Debugf("valid ingress suffix detected, using ingress class name %s", ingressClass)
+	if ingressName, ok := loadBalancers[strings.Split(ingressClass, ".")[0]]; ok {
+		logger.Debugf("valid ingress name detected %s, using ingress class %s", ingressName, ingressClass)
```
Comment (maximilien) [inline pkg/reconciler/ingress/ingress.go:72]: I would not leave debug logs in the code. Find for info and your debugging but not to leave there in official PR

**Review by maximilien — COMMENTED** [review]
Left some comments

**Review by maximilien — APPROVED** [review]
/approve
/lgtm

**Pull request merged**